_posts/2020-08-19-openapi-for-everyone.adoc

```diff
@@ -0,0 +1,843 @@
+----
+
+---
+openapi: 3.0.3
+info:
+  title: OpenAPI for Everyone (development)
+  description: Example on how to use OpenAPI everywhere
+  contact:
+    name: Phillip Kruger
+    url: https://www.phillip-kruger.com
+    email: phillip.kruger@redhat.com
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 1.0.0
+
+# Rest of the schema document...
+
+----
+
+=== Adding some OpenAPI Annotations to your operations
+
+You can use any of the annotations in MicroProfile OpenAPI to further describe your endpoint, for example the `Tag` annotation:
+
+[source, java]
+----
+
+@Path("/jax-rs")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Tag(name = "JAX-RS Resource", description = "Basic Hello World using JAX-RS") // <1>
+public class JaxRsGreeting {
+    //...
+}
+
+----
+<1> Example usage of MicroProfile OpenAPI annotation.
+
+=== Auto generate the operation id
+
+Some tools that use the schema document to generate client stubs, need an `operationId` in the schema document that is used to name the client stub methods.
+We added support in SmallRye to auto generate this using either the method name (`METHOD`), class and method name (`CLASS_METHOD`), or package, class and method name (`PACKAGE_CLASS_METHOD`).
+To do this add the following to `application.properties`:
+
+[source, properties]
+----
+
+mp.openapi.extensions.smallrye.operationIdStrategy=METHOD
+
+----
+
+You will now see the `operationId` in the schema document for every operation:
+
+[source, yaml]
+----
+
+---
+openapi: 3.0.3
+
+# Header omitted...
+
+/jax-rs/hello:
+    get:
+      tags:
+      - JAX-RS Resource
+      operationId: helloJaxRs // <1>
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Greeting'
+    post:
+      tags:
+      - JAX-RS Resource
+      operationId: newHelloJaxRs // <1>
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Greeting'
+      responses:
+        "200":
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Greeting'
+  /jax-rs/hello/{message}:
+    delete:
+      tags:
+      - JAX-RS Resource
+      operationId: deleteHelloJaxRs // <1>
+      parameters:
+      - name: message
+        in: path
+        required: true
+        schema:
+          type: string
+      responses:
+        "204":
+          description: No Content
+
+----
+<1> Auto generated `operationId`
+
+=== Changing the OpenAPI version
+
+Some API gateways might require a certain OpenAPI version to work. The schema document generated by the SmallRye extension
+will generate with a `3.0.3` as the version, but since there is only minor differences between these versions, you can change that to
```

Comment:
```suggestion
will generate with `3.0.3` as the version, but since there is only minor differences between these versions, you can change that to
```